resolver/dns_resolver.ml

```diff
@@ -284,6 +284,36 @@ let likely_blocked reply =
   | `Rcode_error (Rcode.NXDomain, _, Some (_answ, auth)) -> blocked_soa auth
   | _ -> false
 
+let blocking_reason reply =
+  let find_soa_hostmaster rr =
+    let soa = Rr_map.find Rr_map.Soa rr in
+    Option.map (fun soa -> Domain_name.to_string soa.Soa.hostmaster) soa
+  in
```

Comment:
In `likely_blocked` we use a heuristic that the soa has a nameserver of either `localhost` or `blocked`. Maybe we should filter on that here too?

Reply:
see [babc797](https://github.com/mirage/ocaml-dns/pull/384/commits/babc7974ab98adade32d9d844785a5b5f306211b)